vpr/src/pack/re_cluster.cpp

```diff
@@ -193,6 +203,11 @@ bool swap_two_molecules(t_pack_molecule* molecule_1,
         return false;
     }
 
+    t_pb* clb_pb_1 = cluster_ctx.clb_nlist.block_pb(clb_1);
+    std::string clb_pb_1_name = (std::string)clb_pb_1->name;
+    t_pb* clb_pb_2 = cluster_ctx.clb_nlist.block_pb(clb_2);
+    std::string clb_pb_2_name = (std::string)clb_pb_2->name;
```

Comment:
I am not familiar with the style guidelines of VTR; however, these are C style casts. I would recommend changing them to c++ style casts (static_cast<std::string>). This just allows the compiler to check for issues at compile time and they just look good. https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting

Reply:
However, if the t_pb struct / class's name member is a char *, it may be better to use the std::string constructor.

std::string clb_pb_1_name(clb_pb_1->name);

for example.